server/src/jobs/oneshot/afFormation/importer.js

```diff
@@ -204,6 +206,52 @@ const seed = async () => {
   logger.info(`${count} formations importées !`);
 };
 
+const update_oleoduc = async (tableCorrespondance) => {
+  let count = 0;
+
+  await oleoduc(
+    AfFormation.find({ libelle_ban: { $ne: null }, code_cfd: { $eq: null } }).cursor(),
+    writeData(
+      async (formation) => {
+        const mef10 = formation.code_mef.slice(0, -1);
+        const isValid = isFinite(parseInt(mef10));
+
+        let code_cfd;
+
+        if (isValid) {
+          const cfdFromTCO = await getCfdFromTCO(tableCorrespondance, mef10);
+
+          if (cfdFromTCO) {
+            code_cfd = cfdFromTCO;
+          }
+        } else {
+          const cfdFromCatalogue = await getCfdFromCatalogue(tableCorrespondance, formation);
+
+          if (cfdFromCatalogue) {
+            code_cfd = cfdFromCatalogue;
+          } else {
+            const cfdFromBCN = await getCfdFromBCN(tableCorrespondance, formation.libelle_ban, formation.type_voie);
+
+            if (cfdFromBCN) {
+              code_cfd = cfdFromBCN;
+            }
+          }
+        }
+
+        if (!code_cfd) return;
+
+        logger.info(`MAJ formation ${formation._id} - cfd : ${code_cfd}`);
+        await AfFormation.findByIdAndUpdate(formation._id, { code_cfd });
```

Comment:
I guess, you may just `save()` the formation here ?

Reply:
I'm not 100% sure of the formation shape here, also i'd have to update the code to save directly inside the formation object i guess, so I'm adding lean() to the query to minimize the payload.